### PR TITLE
Pass ignore special file names

### PR DESCRIPTION
**Pull request by chezou**
In some Hugo theme like Wowchemy, there is a special index page to list
all of the section, that is usually named as `_index.md`.

This patch enables to ignore those special file names in Handler.

### CODE REVIEW

File: prelims/processor/tests/test_recommender.py [hunk 1]
```diff
@@ -4,18 +4,15 @@
...
         # file paths
-        self.path_a = os.path.abspath(
-                os.path.join(os.sep, 'path', 'to', 'articles', 'a.md'))
-        self.path_b = os.path.abspath(
-                os.path.join(os.sep, 'path', 'to', 'articles', 'b.md'))
-        self.path_c = os.path.abspath(
-                os.path.join(os.sep, 'path', 'to', 'articles', 'c', 'index.md'))
+        self.path_a = Path('path').joinpath(f'{os.sep}to', 'articles', 'a.md')
+        self.path_b = Path('path').joinpath(f'{os.sep}to', 'articles', 'b.md')
+        self.path_c = Path('path').joinpath(f'{os.sep}to', 'articles', 'c', 'index.md')
```
Comment (takuti) [inline prelims/processor/tests/test_recommender.py]: Do you mean `Path(f'{os.sep}path').joinpath('to', 'articles', 'a.md')`?

```py
>>> from pathlib import Path
>>> import os
>>> Path('path').joinpath(f'{os.sep}to', 'articles', 'a.md')
PosixPath('/to/articles/a.md')
>>> Path(f'{os.sep}path').joinpath('to', 'articles', 'a.md')
PosixPath('/path/to/articles/a.md')
```

Also appreciated if you could update https://github.com/takuti/prelims/blob/master/prelims/processor/tests/test_opengraph.py#L12-L13 accordingly.

File: prelims/processor/recommender.py [hunk 2]
```diff
@@ -108,7 +108,7 @@ def __path_to_permalink(self, path):
         """Convert a file path into a permalink, which is a part of final URL
         excluding a file extension.
         """
-        file, _ = os.path.splitext(os.path.basename(path))
+        file = path.stem
         if file == 'index':
-            file = os.path.basename(os.path.dirname(path))
+            file = path.parent.name
```
Comment (takuti) [inline prelims/processor/recommender.py:113]: Nice catch on the difference in type: `str` vs. `Path`. Thank you.

File: prelims/handler.py [hunk 3]
```diff
@@ -34,6 +35,9 @@ def load_posts(self):
         """
         posts = []
         for path in self.paths:
+            if path.name in self.ignore_files:
+                continue
```
Comment (takuti) [inline prelims/handler.py:39]: Can we test this in `prelims/tests/test_handler.py`?

Comment (chezou) [reply prelims/handler.py:39]: Added 1f9c892

File: prelims/processor/tests/test_recommender.py (hunk 1 above)
Comment (chezou) [reply prelims/processor/tests/test_recommender.py]: Fixed 6563918

**Comment by takuti** [comment]
LGTM, thank you! Would it be possible for you to cleanse the commit logs wth `git rebase maser`?

**Comment by chezou** [comment]
Rebased, there you go

**Pull request merged**